src/flask_cognito_lib/services/cognito_svc.py

```diff
@@ -96,11 +96,16 @@ def exchange_code_for_token(
             "code_verifier": code_verifier,
         }
 
+        if self.cfg.user_pool_client_secret:
+            auth = (self.cfg.user_pool_client_id, self.cfg.user_pool_client_secret)
+        else:
+            auth = None
```

Comment:
It's possible this change will reduce code coverage (need to check why the CI pipeline didn't run against this PR). Please could you take a look at adding some test coverage to `tests/test_cognito_svc.py`?

Reply:
Please could you merge latest changes from master, should run the CI checks against this PR. thanks

Reply:
Sorry but for the test coverage, I might need some advice:
I can see in `tests/conftest.py` we create the app with `_app.config["AWS_COGNITO_USER_POOL_CLIENT_SECRET"] = "secure-client-secret"` and then define the `pytest.fixture` `cfg` using that.

I want to run the same tests in `tests/test_cognito_svc.py` but without the secret being set, but I can't quite see the best way to do it yet.

I'm thinking the correct approach is some sort of `cfg_no_secret` fixture with the `user_pool_client_secret` property patched? And then another set of test calls that use this?

Reply:
Hi, I think I have this correct now, it took a few attempts though :P

![image](https://github.com/mblackgeo/flask-cognito-lib/assets/44370243/4d62c5a5-fc6b-4e72-8da1-ed0cdd0c61d5)

Let me know if this is ok, thanks!